app/main/views/suppliers.py

```diff
@@ -376,32 +377,49 @@ def view_signed_agreement(supplier_id, framework_slug):
     if not supplier_framework.get('agreementReturned'):
         abort(404)
 
+    lot_names = []
     if framework["status"] in ("live", "expired"):
         # If the framework is live or expired we don't need to filter drafts, we only care about successful services
         service_iterator = data_api_client.find_services_iter(supplier_id=supplier_id, framework=framework_slug)
-        lot_slugs_names = [(service["lotSlug"], service["lotName"]) for service in service_iterator]
+        for service in service_iterator:
+            if service['lotName'] not in lot_names:
+                lot_names.append(service['lotName'])
     else:
         # If the framework has not yet become live we need to filter out unsuccessful services
         service_iterator = data_api_client.find_draft_services_iter(supplier_id=supplier_id, framework=framework_slug)
-        lot_slugs_names = [
-            (service["lotSlug"], service["lotName"]) for service in service_iterator if service["status"] == "submitted"
-        ]
+        for service in service_iterator:
+            if service["status"] == "submitted" and service['lotName'] not in lot_names:
+                lot_names.append(service['lotName'])
 
     agreements_bucket = s3.S3(current_app.config['DM_AGREEMENTS_BUCKET'])
-    path = supplier_framework['agreementPath']
-    url = get_signed_url(agreements_bucket, path, current_app.config['DM_ASSETS_URL'])
+
+    if framework_slug in OLD_COUNTERSIGNING_FLOW_FRAMEWORKS:
+        is_e_signature_flow = False
+        # Fetch path to supplier's signature page
+        path = supplier_framework['agreementPath']
+        template = "suppliers/view_signed_agreement.html"
+    else:
+        is_e_signature_flow = True
+        # Fetch path to combined countersigned agreement, if available
+        path = supplier_framework.get('countersignedPath', "")
```

Comment:
What's the effect of the `""` parameter here?

Reply:
It sets the default fallback, so instead of `None` we have a string. I don't think we actually need it any more, as we're doing an existence check below. Happy to remove?

Reply:
@katstevens makes sense to remove I think, if subsequent behaviour is the same without it.

Reply:
@gidsg Now done